examples/hitl/rearrange_v2/ui.py

```diff
@@ -337,6 +343,234 @@ def _update_held_object_placement(self) -> None:
+            v = np.where(t6, 1, v)
+            u = np.where(t7, np.clip(numer1, 0, 1), u)
+            v = np.where(t7, 1 - u, v)
+            u = np.where(t8, 1, u)
+            v = np.where(t8, 0, v)
+            u = np.where(t9, np.clip(numer1, 0, 1), u)
+            v = np.where(t9, 1 - u, v)
+            u = u[:, None]
+            v = v[:, None]
+
+            # this array contains a list of points, the closest on each triangle
+            closest_points_each_tri = p0 + u * e0 + v * e1
+
+            # now extract the closest point on the mesh and minimum distance for return
+            closest_point_index = np.argmin(
+                spatial.distance.cdist(
+                    np.array([point]), closest_points_each_tri
+                ),
+                axis=1,
+            )
+            closest_point: np.ndarray = closest_points_each_tri[
+                closest_point_index
+            ]
+            min_dist = float(np.linalg.norm(point - closest_point))
+
+            # Return the minimum distance
+            return min_dist, closest_point
+
+    def compute_dist_to_recs(
+        self, point: np.ndarray, candidate_recs: List[Receptacle]
+    ) -> List[float]:
+        """
+        For each receptacle in the input list, compute a distance from point to receptacle and return the list of distances.
+
+        :param point: A 3D point in global space. Typically the bottom center point of a placed object.
+        :param candidate_recs: A list of candidate Receptacles which could be matched to the point. Typically a subset of all Receptacles.
+        :return: A list of point to Receptacle distances, one for each input in candidate_recs .
+        """
+
+        dist_to_recs = []
+        for rec in candidate_recs:
+            if isinstance(rec, TriangleMeshReceptacle):
+                t_form = rec.get_global_transform(self._sim)
+                # optimization: transform the point into local space instead of transforming the mesh into global space
+                local_point = t_form.inverted().transform_point(point)
+                # iterate over the triangles, getting point to edge distances
+                # NOTE: list of lists, each with 3 numpy arrays, one for each vertex
+                # TODO: these could be cached since it doesn't require local->global transform
+                triangles = []
+                for f_ix in range(int(len(rec.mesh_data.indices) / 3)):
+                    v = rec.get_face_verts(f_ix)
+                    triangles.append(v)
+                np_tri = np.array(triangles)
+                np_point = np.array(local_point)
+                # compute the minimum point to mesh distance
+                p_to_t_dist = self.point_to_tri_dist(np_point, np_tri)[0]
+                dist_to_recs.append(p_to_t_dist)
+            else:
+                raise NotImplementedError(
+                    "TODO: add handling for other Receptacle types."
+                )
```

Comment:
Note the limitation. This will only be a problem for use with ReplicaCAD and other older AABBReceptacle types.